core/src/main/java/com/scalar/db/storage/jdbc/query/SelectWithTop.java

```diff
@@ -3,18 +3,18 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
-public class SelectWithOffsetFetchQuery extends SimpleSelectQuery {
+public class SelectWithTop extends SimpleSelectQuery {
 
   private final int limit;
 
-  public SelectWithOffsetFetchQuery(Builder builder, int limit) {
+  public SelectWithTop(Builder builder, int limit) {
     super(builder);
     assert limit > 0;
     this.limit = limit;
   }
 
   @Override
   public String sql() {
-    return super.sql() + " OFFSET 0 ROWS FETCH NEXT " + limit + " ROWS ONLY";
+    return new StringBuilder(super.sql()).insert(7, "TOP " + limit + " ").toString();
```

Comment:
Haven't checked the other code, but the hard-coded `7` is always OK?

Reply:
Yes. 
The `super.sql()` always returns a select query like `SELECT ...`. This method inserts right after the `SELECT` the `TOP X` option. 
I used the `StringBuilder.insert` instead of something more readable like a `String.replace()` because it is less CPU intensive.

Reply:
Ah, OK. I didn't really know "SELECT TOP` clause. Thank you!
Maybe better to add a comment on the above to specify what `7` means for people who don't know the `SELECT TOP` clause like me. :)

Reply:
Sure 🙂